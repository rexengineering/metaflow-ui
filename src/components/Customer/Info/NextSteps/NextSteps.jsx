import React from "react";
import PropTypes from "prop-types";
import InfoSection from "../InfoSection";
import ActionItem from "../../../ActionItem";
import { ActionItemCheckType } from "../../../../constants";

function NextSteps({
  nextSteps,
  className,
  onAddNextStepsButtonClicked,
  onNextStepsSelected,
}) {
  return (
    <InfoSection
      className={className}
      onAddIconClicked={onAddNextStepsButtonClicked}
      title="Next Steps"
    >
      {Array.isArray(nextSteps) &&
        nextSteps.map((nextStep) => (
          <ActionItem
            key={nextStep}
            label={nextStep}
            actionType={ActionItemCheckType}
            onAction={onNextStepsSelected}
          />
        ))}
    </InfoSection>
  );
}

NextSteps.defaultProps = {
  className: "",
};

NextSteps.propTypes = {
  nextSteps: PropTypes.arrayOf(PropTypes.string).isRequired,
  className: PropTypes.string,
  onAddNextStepsButtonClicked: PropTypes.func.isRequired,
  onNextStepsSelected: PropTypes.func.isRequired,
};

export default NextSteps;
