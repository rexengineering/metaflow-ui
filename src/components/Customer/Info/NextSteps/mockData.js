export default ["Schedule an IHC", "Schedule an XXXXX"];
