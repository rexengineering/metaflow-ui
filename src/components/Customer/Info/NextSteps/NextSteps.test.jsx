import { render, screen } from "@testing-library/react";
import React from "react";
import NextSteps from ".";
import nextStepsMock from "./mockData";

describe("<NextSteps />", () => {
  const renderNextStepsComponent = (
    onAddNextStepsButtonClicked = () => {},
    onNextStepsSelected = () => {}
  ) => {
    return (
      <NextSteps
        onAddNextStepsButtonClicked={onAddNextStepsButtonClicked}
        onNextStepsSelected={onNextStepsSelected}
        nextSteps={nextStepsMock}
      />
    );
  };

  it("should render", () => {
    const { asFragment } = render(renderNextStepsComponent());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(renderNextStepsComponent());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should render all passed items", () => {
    render(renderNextStepsComponent());
    nextStepsMock.forEach((nextStep) => {
      const nextStepNode = screen.getByText(nextStep);
      expect(nextStepNode).toBeInTheDocument();
    });
  });
});
