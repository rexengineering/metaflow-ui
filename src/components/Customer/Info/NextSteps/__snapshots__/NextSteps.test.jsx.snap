// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NextSteps /> should match snapshot 1`] = `
<DocumentFragment>
  <section
    class=""
  >
    <div
      class="makeStyles-header-11"
    >
      <h6
        class="MuiTypography-root makeStyles-label-12 MuiTypography-h6"
      >
        Next Steps
      </h6>
      <button
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-plus-circle fa-w-16 "
            data-icon="plus-circle"
            data-prefix="fal"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 250v12c0 6.6-5.4 12-12 12h-98v98c0 6.6-5.4 12-12 12h-12c-6.6 0-12-5.4-12-12v-98h-98c-6.6 0-12-5.4-12-12v-12c0-6.6 5.4-12 12-12h98v-98c0-6.6 5.4-12 12-12h12c6.6 0 12 5.4 12 12v98h98c6.6 0 12 5.4 12 12zm120 6c0 137-111 248-248 248S8 393 8 256 119 8 256 8s248 111 248 248zm-32 0c0-119.9-97.3-216-216-216-119.9 0-216 97.3-216 216 0 119.9 97.3 216 216 216 119.9 0 216-97.3 216-216z"
              fill="currentColor"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div>
      <div
        class="makeStyles-container-13"
      >
        <p
          class="MuiTypography-root makeStyles-label-14 MuiTypography-body1"
        >
          Schedule an IHC
        </p>
        <div
          class="makeStyles-actionElement-15"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-20"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
      </div>
      <div
        class="makeStyles-container-13"
      >
        <p
          class="MuiTypography-root makeStyles-label-14 MuiTypography-body1"
        >
          Schedule an XXXXX
        </p>
        <div
          class="makeStyles-actionElement-15"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-17 MuiCheckbox-root MuiCheckbox-colorSecondary MuiIconButton-colorSecondary"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                class="PrivateSwitchBase-input-20"
                data-indeterminate="false"
                type="checkbox"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
